#working 
7th MAr
uplodaing desktop_apod.py to the GitHub
working with vs_code_ignore file

time:30 mins

24th march

uplodaing uplodaing desktop_apod.py in python file format to the GitHub
working with vs_code_ignore file to remove the workspace files
working with database
time:1 hr 30 mins
reference : lab 3

30th march

worked on the getting apod_info using the NASA APOD API
time:around half an hour
reference:lab 6
https://www.youtube.com/watch?v=cZovlpGVVdA

5th April
Worked on creating the create_image_db function
worked pefectly and I am able to see the database file
time:35 mins

worked on image url, which I get through the apod_info dictionary
reference: lab 7
time:5 mins

worked on the downloading the image from the URL,which I get from the apod_info dictionary
facing the problem to differentiate between the downloadingand the saving fucntion of the APOD script.
reference:https://stackoverflow.com/questions/13137817/how-to-download-image-using-requests
time: approxe 47 mins

worked on the saving the image file on the disk.
Same code with the as the last edition but with the bit correction and repleacements.
time:5 mins

worked on the image hash
first couter some problem when I was doing with the fucntion, which was couting the hash of the image
problem:function somehow fist gave me error and then I edited it little bit but still it wasn't giving me error but giving me garbage value for hash

Then instead of using the function I just implement the code into the main() function
reference:https://www.quickprogrammingtips.com/python/how-to-calculate-sha256-hash-of-a-file-in-python.html
time: approxe 45 mins

worked on the getting image size
reference:https://stackoverflow.com/questions/2104080/how-can-i-check-file-size-in-python
time:20 mins

worked on the printing the APOD information
time:5 mins

worked on the get_image_path()function:
reference:https://www.geeksforgeeks.org/python-string-rsplit-method/
time:approxe half hour

Try to work on the script execution without saving images into databse and without even putting on desktop background
Error: File path for image_saving is not exsisted
time:approxe 40 mins

6-APR-2022
Able to solve the File_Path problem
solution: I was not calling the save_image_file() function in main(), that's why it wasn't downloading the image and path was showing as unreachable
time:20 mins

worked on adding images into databse
first error in datetime.time() function for passing the date, but then I try datetime.now() which worked perfectly
reference: Lab 3
https://datatofish.com/create-database-python-using-sqlite3/
https://pynative.com/python-sqlite-date-and-datetime/
time:28 mins


